tests/cases/conformance/salsa/inferringClassMembersFromAssignments.ts

```diff
@@ -0,0 +1,79 @@
+// @out: output.js
+// @allowJs: true
+// @strictNullChecks: true
+
+// @filename: a.js
+class C {
+    constructor() {
+        if (Math.random()) {
+            this.inConstructor = 0;
```

Comment:
can you add another property `this.inBoth` that gets `this.inBoth = 0` in the constructor and `this.inBoth = 'string'` in `method()`. The resulting type should be `number | string` I think.